### PR TITLE
Some tests for url predicates and operations

### PR DESCRIPTION
**Pull request by darcy-shen**
What
+ Unit tests on URL predicates and operations
+ Remove is_rooted_path and is_path

How to test locally:

```
cmake .. -DCMAKE_INSTALL_PREFIX=/tmp/software -DBUILD_TESTS=on
make -j8
tests/url_test
```

### CODE REVIEW

File: tests/System/Classes/url_test.cpp [hunk 1]
```diff
@@ -15,33 +15,77 @@
...
+void TestURL::test_is_rooted_web () {
+  QVERIFY (is_rooted_web (http_1));
+  QVERIFY (is_rooted_web (https_1));
+  QVERIFY (!is_rooted_web (root_tmp));
+  QVERIFY (!is_rooted_web (tmfs_1));
+}
+
+void TestURL::test_is_path () {
+  // TODO: is_path is confusing, should be removed?
+  QVERIFY (!is_path (root_tmp));
```
Comment (PikachuHy) [inline tests/System/Classes/url_test.cpp:76]: why `/tmp` is not a path?

Comment (PikachuHy) [reply tests/System/Classes/url_test.cpp:76]: while `tmp` is a path

Comment (PikachuHy) [reply tests/System/Classes/url_test.cpp:76]: the `is_path` is only used by `is_rooted_path`

`is_rooted_path` is not used

Comment (PikachuHy) [reply tests/System/Classes/url_test.cpp:76]: I think it's ok to remove both `is_path` and `is_rooted_path`.

Comment (darcy-shen) [reply tests/System/Classes/url_test.cpp:76]: yes. They should be removed!

**Review by PikachuHy — APPROVED** [review]
LGTM

**Comment by darcy-shen** [comment]
As I mentioned in #130 , we should always use `Squash and merge`.

The owner of the pull requests should rebase his pull request manually.

**Comment by PikachuHy** [comment]
> The owner of the pull requests should rebase his pull request manually.

I can't agree more.

That's what I want to say.

**Pull request merged**